torch/_logging/_internal.py

```diff
@@ -234,16 +239,12 @@ def getArtifactLogger(module_qname, artifact_name):
 
 
 def configure_artifact_log(log):
-    # If the artifact is off by default, then it should only be logged when explicitly
-    # enabled; set propagate to False so that this artifact is not propagated
-    # to its ancestor logger
-    if log_registry.is_off_by_default(log.artifact_name):
-        log.propagate = False
-
     # enable artifact logging when explicitly enabled
     if log_state.is_artifact_enabled(log.artifact_name):
         log.setLevel(logging.DEBUG)
         log.propagate = True
+    else:
+        log.propagate = False
```

Comment:
This is incorrect - before it was known at import time (when this is called) whether or not propagate should be True. Now we have to set this when we know the setting from the user. If the artifact is set to False, propagate should be False, in all other cases it should be True.